tests/test_models_dbi.py

```diff
@@ -40,16 +45,271 @@ def test_double_bracket_iteration_group_commutator(backend, nqubits):
 
     # test first iteration with default d
     dbi(mode=DoubleBracketGeneratorType.group_commutator, step=0.01)
-
     for _ in range(NSTEPS):
         dbi(step=0.01, d=d)
 
     assert initial_off_diagonal_norm > dbi.off_diagonal_norm
 
 
+@pytest.mark.parametrize("nqubits", [3])
+def test_double_bracket_iteration_eval_dbr_unitary(backend, nqubits):
+    r"""The bound is $$||e^{-[D,H]}-GC||\le s^{3/2}(||[H,[D,H]||+||[D,[D,H]]||$$ which we check by a loglog fit."""
+    h0 = random_hermitian(2**nqubits, backend=backend)
+    d = backend.cast(np.diag(np.diag(backend.to_numpy(h0))))
+    dbi = DoubleBracketIteration(
+        Hamiltonian(nqubits, h0, backend=backend),
+        mode=DoubleBracketGeneratorType.group_commutator,
+    )
+
+    times = np.linspace(0.001, 0.01, 10)
+    norms = []
+    norms_bound = []
+    for s in times:
+        u = dbi.eval_dbr_unitary(
+            s, d=d, mode=DoubleBracketGeneratorType.single_commutator
+        )
+        v = dbi.eval_dbr_unitary(
+            s, d=d, mode=DoubleBracketGeneratorType.group_commutator
+        )
+
+        norms.append(np.linalg.norm(u - v) )
+        w = dbi.commutator(h0,d)
+        norms_bound.append(0.5*s**1.48 * (
+            np.linalg.norm(dbi.commutator(h0,w)) + np.linalg.norm(dbi.commutator(d,w))
+        ))
+        assert np.linalg.norm(u - v) < 10 * s**1.49 * (
+                np.linalg.norm(h0) + np.linalg.norm(d)
+            ) * np.linalg.norm(h0) * np.linalg.norm(d)
+
+
+
+@pytest.mark.parametrize("nqubits", [3])
+def test_dbi_evolution_oracle(backend, nqubits, t_step = 0.1, eps = 0.001 ):
```

Comment:
@Edoardo-Pedicillo Is this the lint complaint? I.e. the default inits are not allowed?